File: 0013-roman-to-integer/0013-roman-to-integer.py
romanNums = {"M": 1000, "D": 500, "C": 100, "L": 50, "X": 10, "V": 5, "I": 1}


class Solution:
    def romanToInt(self, s: str) -> int:
        result = 0
        for i in range(len(s) - 1):
            if romanNums[s[i]] < romanNums[s[i + 1]]:
                result = result - romanNums[s[i]]
            else:
                result = result + romanNums[s[i]]
        result = result + romanNums[s[len(s) - 1]]
        return result

File: 0013-roman-to-integer/0013-roman-to-integer.java
class Solution {
    public int romanToInt(String s) {
        Map<Character, Integer> romanCharMap = new HashMap<>();
        romanCharMap.put('I', 1);
        romanCharMap.put('V', 5);
        romanCharMap.put('X', 10);
        romanCharMap.put('L', 50);
        romanCharMap.put('C', 100);
        romanCharMap.put('D', 500);
        romanCharMap.put('M', 1000);
        
        int totalVal = 0;

        for (int i = 0; i < s.length() - 1; i++) {
            if (romanCharMap.get(s.charAt(i)) < romanCharMap.get(s.charAt(i + 1))) {
                totalVal -= romanCharMap.get(s.charAt(i));
            }
            else{
                totalVal += romanCharMap.get(s.charAt(i));
            }
        }
        totalVal += romanCharMap.get(s.charAt(s.length() - 1));
        return totalVal;
    }
}


